Sources/Network/OfflineProductsService.swift

```diff
@@ -140,10 +142,32 @@ class OfflineProductsService: OfflineProductsApi {
         }
     }
 
+    private var forceRefresh: Bool {
+        // The last refresh date in seconds since 1970
+        let currentOfflineProductsDate = Date(timeIntervalSince1970: UserDefaults.standard.double(forKey: OfflineProductsService.USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD))
+        // The refresh date to use 31-dec-2020 24h GMT gregorian
+        var dateComponents = DateComponents()
+        dateComponents.year = 2020
+        dateComponents.month = 12
+        dateComponents.day = 31
+        dateComponents.hour = 24
+        dateComponents.minute = 0
+        dateComponents.second = 0
+        dateComponents.calendar = Calendar(identifier: .gregorian)
+        dateComponents.timeZone = TimeZone.init(secondsFromGMT: 0)
```

Comment:
A bit verbose... not truly an issue as this code is temporary (we might add a TODO: remove after 1st january 2020), but `Date(timeIntervalSince1970: 1609455599)` with a comment would have been clearer IMO.

Apart from this, what is the wanted behaviour for this ? Why force a refresh on 1st january? (I probably miss some insights on the release of the feature or something like that)

Reply:
We're going to ship a version with ecoscore_alpha to internal (that activates early Eco-Score display). If everything goes well, we're shipping a version without ecoscore_alpha to the whole userbase (to start packaging collection). 
There's no set date for the release (we had a meeting and a tentative date was set, but that did not inform the January 1st thing)

Reply:
So I don't understand when we want the "force refresh" behaviour? If we want to force refresh the data when the user makes the update (containing the display of ecoscore), then we should just change the `OfflineProductsService.USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD` constant (adding a suffix `_2` would suffice), so that there is no last update date saved in the UserDefaults, and the refresh will be done.

Reply:
1609455599 when is that? That is why I took the complex route. 
The date is to be set later, when there is a launch date for ecoscore.
Yes, we could add another user default, but I thought this approach would be more cleaner and could be reused in the future, when other things are added.

Reply:
1609455599 is the same date, with a comment above it is fine. The longuest route is just 8 lines and therefore not clearer (still IMO, it is not really improtant).
For the refresh I'm talking about just changing the value of the existing user default key.

See https://github.com/openfoodfacts/openfoodfacts-ios/blob/4f7b2da90cdb49709db79e81b7f0610b37d1888d/Sources/Network/OfflineProductsService.swift#L47 :
Currently we have `OfflineProductsService.USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD = "USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD__13 "`
Changing the const to  `OfflineProductsService.USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD = "USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD__14"` would be enough to trigger a full refresh when the user update the app.

But again, I think I do not understand the release process you want for this feature. Should it appear dynamically after a certain date (that will get decided later), or should it appear when user make the update ?
Also, will the offline dataset incorporate those new fields after a certain date, or do they already incorporate them ? 

If the server is already sending data, then we should just bump the value of the const as explained below so that people get new data when updating the app (whenever they do).
If we want to **display** those new data **after** a certain date, then I think we should go for a "feature flag" only activating the display, not the download. (because if the server is already sending the new fields, then user downloading the app for the first time will download the latest dataset and then see the data).

Reply:
I think `USER_DEFAULT_LAST_OFFLINE_PRODUCTS_DOWNLOAD__14` will be sufficient then, as the new ecoscore data access will correspond to the a new app version.

Reply:
implemented